pkg/resource/clickpipe.go

```diff
@@ -896,69 +1083,99 @@ func (c *ClickPipeResource) Create(ctx context.Context, request resource.CreateR
 	}
 
 	clickPipe.Destination = api.ClickPipeDestination{
-		Database:     destinationModel.Database.ValueString(),
-		Table:        destinationModel.Table.ValueString(),
-		ManagedTable: destinationModel.ManagedTable.ValueBool(),
-		Columns:      make([]api.ClickPipeDestinationColumn, len(destinationColumnsModels)),
-		Roles:        rolesSlice,
+		Database: destinationModel.Database.ValueString(),
+		Roles:    rolesSlice,
 	}
 
-	if destinationModel.ManagedTable.ValueBool() {
-		if destinationModel.TableDefinition.IsNull() {
+	// For non-Postgres sources, set table, managedTable, and columns
+	if !isPostgresSource {
+		// Validate that required fields are provided
+		if destinationModel.Table.IsNull() || destinationModel.Table.ValueString() == "" {
+			response.Diagnostics.AddError(
+				"Error Creating ClickPipe",
+				"destination.table is required for non-Postgres CDC sources.",
+			)
+			return
+		}
+
+		if destinationModel.Columns.IsNull() || len(destinationModel.Columns.Elements()) == 0 {
 			response.Diagnostics.AddError(
 				"Error Creating ClickPipe",
-				"Managed table requires table definition",
+				"destination.columns is required for non-Postgres CDC sources.",
```

Comment:
I'd rather put a positive rather than negated error message.
`destination.columns is required for 's3' source.`